pkgs/development/python-modules/dukpy/default.nix

```diff
@@ -0,0 +1,30 @@
+{ lib
+, fetchPypi
+, buildPythonPackage
+, setuptools
+, mutf8
+}:
+
+buildPythonPackage rec {
+  pname = "dukpy";
+  version = "0.4.0";
+  pyproject = true;
+
+  src = fetchPypi {
+    inherit pname version;
+    hash = "sha256-Z37HEC0cHFEffvkYB46AmXeNvOp8rz1qKipy9yqi0TU=";
+  };
+
+  nativeBuildInputs = [ setuptools ];
+
+  propagatedBuildInputs = [ mutf8 ];
+
+  pythonImportsCheck = [ "dukpy" ];
+
```

Comment:
Could you enable tests with `pytestCheckHook`?
https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#using-pytestcheckhook-using-pytestcheckhook

Reply:
I'm getting an ImportError that I don't understand:
```py
python3.12-dukpy> ____________________ ERROR collecting tests/test_evaljs.py _____________________
python3.12-dukpy> ImportError while importing test module '/build/source/tests/test_evaljs.py'.
python3.12-dukpy> Hint: make sure your test modules/packages have valid Python names.
python3.12-dukpy> Traceback:
python3.12-dukpy> /nix/store/z7xxy35k7620hs6fn6la5fg2lgklv72l-python3-3.12.4/lib/python3.12/importlib/__init__.py:90: in import_module
python3.12-dukpy>     return _bootstrap._gcd_import(name[level:], package, level)
python3.12-dukpy> tests/test_evaljs.py:7: in <module>
python3.12-dukpy>     import dukpy
python3.12-dukpy> dukpy/__init__.py:1: in <module>
python3.12-dukpy>     from .evaljs import evaljs, JSInterpreter
python3.12-dukpy> dukpy/evaljs.py:11: in <module>
python3.12-dukpy>     from . import _dukpy
python3.12-dukpy> E   ImportError: cannot import name '_dukpy' from partially initialized module 'dukpy' (most likely due to a circular import) (/build/source/dukpy/__init__.py)
```

Reply:
Adding the following preCheck might be a good idea.
It usually happens because of a problem loading C extension modules.
```nix
preCheck = ''
  rm -r src
  # or
  # cd $out
'';
```

Reply:
thanks, deleting the dukpy directory fixed it :+1: